Is it possible to get range working with slice types?

All the slice types define `EItem = [T]`, since `[T]` is not Sized, `RangeFrom<&[T]>` (and others) do not implement `RangeBounds`.
Is there some workaround that I am missing?

My use case is a simple key like `[u8; N]` or `Vec<u8>`, etc.

I want a delete_prefix, but as there is none I'm trying to use delete_range() but have encountered this issue.
This is on 0.20.0-alpha6.  Here's a snippet:

````rust
// This is the prefix I want to delete
let start_key_slice: &[u8] = id.as_slice();

// Here I make a copy and increment the last byte, so I can make a range
let end_key = id;
end_key.0[end_key.len() - 1] += 1;
let end_key_slice: &[u8] = end_key.as_slice();

// Here I use a let binding to make an owned range so I can reference it
let range = start_key_slice .. end_key_slice;

// And this line gives the error
self.db_event_seen_on_relay()?.delete_range(txn, &range)?;
````

The error is:
````
error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
    --> gossip-lib/src/storage/mod.rs:1231:62
     |
1231 |             self.db_event_seen_on_relay()?.delete_range(txn, &range)?;
     |                                            ------------      ^^^^^^ doesn't have a size known at compile-time
     |                                            |
     |                                            required by a bound introduced by this call
     |
     = help: the trait `Sized` is not implemented for `[u8]`
     = help: the following other types implement trait `RangeBounds<T>`:
               std::ops::Range<&T>
               std::ops::Range<T>
     = note: required for `std::ops::Range<&[u8]>` to implement `RangeBounds<[u8]>`
note: required by a bound in `heed::Database::<KC, DC, C>::delete_range`
    --> /home/mike/.cargo/git/checkouts/heed-cddcbc156c1cdd8e/8bfdf3b/heed/src/database.rs:2092:12
     |
2089 |     pub fn delete_range<'a, 'txn, R>(&self, txn: &'txn mut RwTxn, range: &'a R) -> Result<usize>
     |            ------------ required by a bound in this associated function
...
2092 |         R: RangeBounds<KC::EItem>,
     |            ^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Database::<KC, DC, C>::delete_range`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `gossip-lib` (lib) due to previous error

````

Hi all, I'm trying to implement the solution described above and I'm running into the same issue.

Here's what I have:
```rs
let start_key_slice: &[u8] = req.start.as_slice();
            let end_key_slice: &[u8] = req.end.as_slice();
            let range = (&start_key_slice)..(&end_key_slice);
            for entry in self
                .db
                .range(&txn, &range)
                .map_err(|e| Error::msg(e.to_string()))?
            {}
```

And here's the error:
```
error[E0277]: the trait bound `std::vec::Vec<u8>: heed::BytesDecode<'_>` is not satisfied
  --> src/lib.rs:61:26
   |
61 |               for entry in self
   |  __________________________^
62 | |                 .db
63 | |                 .range(&txn, &range)
64 | |                 .map_err(|e| Error::msg(e.to_string()))?
   | |________________________________________________________^ the trait `heed::BytesDecode<'_>` is not implemented for `std::vec::Vec<u8>`
   |
   = help: the following other types implement trait `heed::BytesDecode<'a>`:
             <heed::heed_types::CowSlice<T> as heed::BytesDecode<'a>>
             <heed::heed_types::CowType<T> as heed::BytesDecode<'a>>
             <heed::heed_types::OwnedSlice<T> as heed::BytesDecode<'a>>
             <heed::heed_types::OwnedType<T> as heed::BytesDecode<'a>>
             <heed::heed_types::Str as heed::BytesDecode<'a>>
             <heed::heed_types::UnalignedSlice<T> as heed::BytesDecode<'a>>
             <heed::heed_types::UnalignedType<T> as heed::BytesDecode<'a>>
             <heed::heed_types::Unit as heed::BytesDecode<'_>>
           and 4 others
   = note: required for `heed::RoRange<'_, std::vec::Vec<u8>, denokv_proto::KvValue>` to implement `std::iter::Iterator`
   = note: required for `heed::RoRange<'_, std::vec::Vec<u8>, denokv_proto::KvValue>` to implement `std::iter::IntoIterator`
```

I'm sure I'm missing something really basic but I'm not super sure what... any help would be appreciated. `req.start` and `req.end` are both Vec<u8>.

--- a/README.md
+++ b/README.md
@@ -6,6 +6,7 @@
 [![Docs](https://docs.rs/heed/badge.svg)](https://docs.rs/heed)
 [![dependency status](https://deps.rs/repo/github/meilisearch/heed/status.svg)](https://deps.rs/repo/github/meilisearch/heed)
 [![Build](https://github.com/meilisearch/heed/actions/workflows/rust.yml/badge.svg)](https://github.com/meilisearch/heed/actions/workflows/rust.yml)
+![Discord](https://img.shields.io/discord/1006923006964154428?style=flat&logo=discord&logoColor=ffffff&label=&labelColor=6A7EC2&color=7389D8&link=https%3A%2F%2Fdiscord.com%2Fchannels%2F1006923006964154428%2F1347203493106024528)
 
 A Rust-centric [LMDB](https://en.wikipedia.org/wiki/Lightning_Memory-Mapped_Database) abstraction with minimal overhead. This library enables the storage of various Rust types within LMDB, extending support to include Serde-compatible types. It not only supports the LMDB `mdb.master` branch but also the `mdb.master3` branch which features encryption-at-rest and checksumming.
 

--- a/heed/src/databases/database.rs
+++ b/heed/src/databases/database.rs
@@ -1231,6 +1231,45 @@ impl<KC, DC, C> Database<KC, DC, C> {
     /// wtxn.commit()?;
     /// # Ok(()) }
     /// ```
+    ///
+    /// It can be complex to work with ranges of slices and using
+    /// the `..` or `..=` range syntax is not the best way to deal
+    /// with that. We highly recommend using the [`Bound`](std::ops::Bound) enum for that.
+    ///
+    /// ```
+    /// # use std::fs;
+    /// # use std::path::Path;
+    /// use std::ops::Bound;
+    /// # use heed::EnvOpenOptions;
+    /// use heed::Database;
+    /// use heed::types::*;
+    /// use heed::byteorder::BigEndian;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// # let dir = tempfile::tempdir()?;
+    /// # let env = unsafe { EnvOpenOptions::new()
+    /// #     .map_size(10 * 1024 * 1024) // 10MB
+    /// #     .max_dbs(30)
+    /// #     .open(dir.path())?
+    /// # };
+    ///
+    /// let mut wtxn = env.write_txn()?;
+    /// let db: Database<Bytes, Unit> = env.create_database(&mut wtxn, None)?;
+    ///
+    /// // make sure to create slices and not ref array
+    /// // by using the [..] syntax.
+    /// let start = &[0, 0, 0][..];
+    /// let end = &[9, 0, 0][..];
+    ///
+    /// // equivalent to start..end
+    /// let range = (Bound::Included(start), Bound::Excluded(end));
+    ///
+    /// let iter = db.range(&mut wtxn, &range)?;
+    ///
+    /// drop(iter);
+    /// wtxn.commit()?;
+    /// # Ok(()) }
+    /// ```
     pub fn range<'a, 'txn, R>(
         &self,
         txn: &'txn RoTxn,

--- a/heed/src/databases/encrypted_database.rs
+++ b/heed/src/databases/encrypted_database.rs
@@ -1064,6 +1064,45 @@ impl<KC, DC, C> EncryptedDatabase<KC, DC, C> {
     /// wtxn.commit()?;
     /// # Ok(()) }
     /// ```
+    ///
+    /// It can be complex to work with ranges of slices and using
+    /// the `..` or `..=` range syntax is not the best way to deal
+    /// with that. We highly recommend using the [`Bound`](std::ops::Bound) enum for that.
+    ///
+    /// ```
+    /// # use std::fs;
+    /// # use std::path::Path;
+    /// use std::ops::Bound;
+    /// # use heed::EnvOpenOptions;
+    /// use heed::Database;
+    /// use heed::types::*;
+    /// use heed::byteorder::BigEndian;
+    ///
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// # let dir = tempfile::tempdir()?;
+    /// # let env = unsafe { EnvOpenOptions::new()
+    /// #     .map_size(10 * 1024 * 1024) // 10MB
+    /// #     .max_dbs(30)
+    /// #     .open(dir.path())?
+    /// # };
+    ///
+    /// let mut wtxn = env.write_txn()?;
+    /// let db: Database<Bytes, Unit> = env.create_database(&mut wtxn, None)?;
+    ///
+    /// // make sure to create slices and not ref array
+    /// // by using the [..] syntax.
+    /// let start = &[0, 0, 0][..];
+    /// let end = &[9, 0, 0][..];
+    ///
+    /// // equivalent to start..end
+    /// let range = (Bound::Included(start), Bound::Excluded(end));
+    ///
+    /// let iter = db.range(&mut wtxn, &range)?;
+    ///
+    /// drop(iter);
+    /// wtxn.commit()?;
+    /// # Ok(()) }
+    /// ```
     pub fn range<'a, 'txn, R>(
         &self,
         txn: &'txn mut RoTxn,

